go/runtime/bundle/manager.go

```
@@ -436,6 +476,32 @@ func (m *Manager) fetchBundle(url string) (string, error) {
 	return file.Name(), nil
 }
 
+func (m *Manager) cleanStaleBundles(runtimeID common.Namespace) {
+	m.mu.Lock()
+	active, ok := m.cleanupQueue[runtimeID]
+	m.mu.Unlock()
+
+	if !ok {
+		return
+	}
+
+	// TODO should you also remove dirs for manifests successfully removed,
```

Whenever you put TODO, you should know who will address this (e.g. you open another PR and assign it to someone, and put the number of that PR to this TODO). Otherwise, sooner or later, there will be TODOs lying all over the place.

Since the store is currently just in-memory store, it doesn't need to return an error. If it does, you should skip removing the bundle as the store still has it in memory and can serve it to its clients. In that case, we don't want to remove the files. We can only do that once manifest was deleted from the memory. So if this fails, maybe it will succeed during the next cleanup.

---

Intentionally left it, so that we can discuss it also opened thread here: https://github.com/oasisprotocol/oasis-core/pull/5976#discussion_r1920398882

Relates to the worry I have above: I.e. `AddManifest` can technically be called concurrently (it is not rn). Race condition might result in manifest being present in the store, but no longer exploded.